src/Symfony/Component/Notifier/Bridge/MicrosoftTeams/Action/MicrosoftTeamsActionCardAction.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\MicrosoftTeams\Action;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+
+/**
+ * @author Edouard Lescot <edouard.lescot@gmail.com>
+ *
+ * @see https://docs.microsoft.com/en-us/outlook/actionable-messages/message-card-reference#actioncard-action
+ */
+class MicrosoftTeamsActionCardAction extends AbstractMicrosoftTeamsAction
+{
+    /**
+     * @return $this
+     */
+    public function name(string $name): self
+    {
+        $this->options['name'] = $name;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function input(MicrosoftTeamsActionInputInterface $inputAction): self
+    {
+        $this->options['inputs'][] = $inputAction->toArray();
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function action(MicrosoftTeamsActionInterface $action): self
+    {
+        if (!\in_array(\get_class($action), [MicrosoftTeamsHttpPostAction::class, MicrosoftTeamsOpenUriAction::class])) {
+            throw new LogicException(sprintf('"%s" must be an instance of "%s" or "%s".', \get_class($action), MicrosoftTeamsHttpPostAction::class, MicrosoftTeamsOpenUriAction::class));
+        }
+
+        $this->options['actions'][] = $action->toArray();
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        $this->options['@type'] = 'ActionCard';
```

Comment:
this is not the right place to change the state. it should be defined in a constructor or merge with the content returned by `parent::toArray`

(several occurrences in this PR)

Reply:
Make sense, will check if this can be done easier (more straight forward)